demos/projects/ESPRESSIF/az-ca-recovery/README.md

```diff
@@ -72,59 +80,32 @@ Parameter | Value
  `WiFi SSID` | _{Your WiFi SSID}_
  `WiFi Password` | _{Your WiFi password}_
 
-
 Under menu item `Azure IoT middleware for FreeRTOS Main Task Configuration`, update the following configuration values:
 
 Parameter | Value
 ---------|----------
- `Use PnP in Azure Sample` | Enabled by default. Disable this option to build a simpler sample without Azure Plug-and-Play.
- `Azure IoT Hub FQDN` | _{Your Azure IoT Hub Host FQDN}_ (Unused if Device Provisioning is enabled below)
- `Azure IoT Device ID` | _{Your Azure IoT Hub device ID}_
- `Azure IoT Module ID` | _{Your Azure IoT Hub Module ID}_ (optional, specify module id if using a device module; else leave blank if not)
+ `Enable Device Provisioning Sample` | _{Check this option to enable DPS in the sample}_
+ `Azure Device Provisioning Service ID Scope` | _{Your ID scope value}_
+ `Azure Device Provisioning Service Registration ID` | _{Your Device Registration ID value}_
+ `[RECOVERY] Azure Device Provisioning Service ID Scope` | _{Your ID scope value for the recovery instance}_
+ `[RECOVERY] Azure Device Provisioning Service Registration ID.` | _{Your Device Registration ID value for the recovery instance}_
 
-Select your desired authentication method with the `Azure IoT Authentication Method () --->`. The default option is `Symmetric Key`:
+Select your desired authentication method with the `Azure IoT Authentication Method () --->` as `Symmetric Key`:
 
 Parameter | Value
 ---------|----------
  `Azure IoT Device Symmetric Key` | _{Your Azure IoT Hub device symmetric key}_
+ `[RECOVERY] Azure IoT Device Symmetric Key` | _{Your Azure IoT Hub device symmetric key for the recovery instance}_
 
-If you would like to use x509 certificates, select `X509 Certificates` and update the following values:
-
-Parameter | Value
----------|----------
- `Azure IoT Device Client Certificate` | _{Your Azure IoT Hub device certificate}_
- `Azure IoT Device Client Certificate Private Key` | _{Your Azure IoT Hub device certificate private key}_
-
-Note that the certificate and private key must be a single line string with `\n` characters at the appropriate line breaks. For example:
+Save the configuration (`Shift + S`) inside the sample folder in a file with name `sdkconfig`.
+After that, close the configuration utility (`Shift + Q`).
 
-```txt
-# PEM Formatted (WRONG)
------BEGIN CERTIFICATE-----
-MIIBJDCBywIUfeHrebBVa2eZAbouBgACp9R3BncwCgYIKoZIzj0EAwIwETEPMA0G
-...
-vTfQahwsxN3xink9z1gtirrjQlqDAiEAyU+6TUJcG6d9JF+uJqsLFpsbbF3IzGAw
-yC+koNRC0MU=
------END CERTIFICATE-----
+You must also update the signing root key which is located in `demo_config.h`, titled `ucAzureIoTRecoveryRootKeyN`. You can get the hex value of the modulus (N value) using the below command.
 
-# Single Line (CORRECT)
------BEGIN CERTIFICATE-----\nMIIBJDCB...\nyC+koNRC0MU=\n-----END CERTIFICATE-----
+```bash
+openssl x509 -in <your-public-cert>.pem -modulus -noout | sed s/Modulus=// | sed -r 's/../0x&, /g'
```

Comment:
Do we use a default exponent of 65537? 
AFAIK RSA will require both N and E.

Reply:
Yes we default to 65537. I can make a note we set it to that by default (and most do), but they might need to update it.